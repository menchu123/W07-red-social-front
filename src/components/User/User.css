.user {
  width: 400px;
  height: 260px;
  background-color: #2e2e2e;
  padding: 30px;
  margin-bottom: 30px;
  box-sizing: border-box;
  border-radius: 5px;
  font-size: 14px;
}
.user__top {
  display: flex;
}
.user__data {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding-bottom: 20px;
}
.user__name {
  text-transform: uppercase;
  font-weight: 700;
}

.user__names {
  display: flex;
  height: 50px;
  flex-direction: column;
  justify-content: space-between;
}

.user__photo {
  object-fit: cover;
  object-position: center;
  margin: 0 20px 20px 0;
}

.friend-button {
  background-color: transparent;
  height: 40px;
  width: 40px;
  border: 3px solid #57554f;
  border-radius: 50%;
  font-size: 16px;
  font-weight: 900;
  color: #57554f;
  transform: rotate(90deg);
  cursor: pointer;
}

.friend-button--friend {
  margin-right: 10px;
}

.friend-button--friend:hover {
  color: #f58eb1;
  border-color: #f58eb1;
}

.friend-button--enemy:hover {
  color: #19decb;
  border-color: #19decb;
}

.enemy-selected {
  background-color: #19decb;
  color: #2e2e2e;
}

.friend-selected {
  background-color: #f58eb1;
  color: #2e2e2e;
}
