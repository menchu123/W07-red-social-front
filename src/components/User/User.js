import "./User.css";

const User = () => {
  return (
    <section className="user">
      <div className="user__top">
        <img
          src="https://cdn.discordapp.com/attachments/896125607170179202/901820331956789248/IMG_3318.jpg"
          alt=""
          className="user__photo"
          height="150"
          width="150"
        />
        <div className="user__data">
          <div className="user__names">
            <span className="user__name">Elsa</span>
            <span className="user__username">@elsithecroc</span>
          </div>

          <div className="friend-buttons">
            <button className="friend-button friend-button--friend">:)</button>
            <button className="friend-button friend-button--enemy">:(</button>
          </div>
        </div>
      </div>
      <div className="user__bio">
        Hola por favor dame comida no he comido nunca por favor se√±or deme de
        comer tengo hambre
      </div>
    </section>
  );
};

export default User;
