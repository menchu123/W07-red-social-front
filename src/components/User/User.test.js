import { render, screen } from "@testing-library/react";
import ReactTestRenderer from "react-test-renderer";

import User from "./User";

describe("Given a User component", () => {
  describe("When it is passed a user object with the name 'Ram贸n'", () => {
    test("Then it should render a card with the user's name", () => {
      const name = "Ram贸n";
      const user = {
        name: name,
        password: "damedecome",
        username: "elsithecroc",
        photo: "",
        bio: "",
      };

      render(<User user={user} />);

      const userName = screen.getByText("Ram贸n");

      expect(userName).toBeInTheDocument();
    });
  });

  describe("When it recives and object", () => {
    test("then it should render a card with the object info inside", () => {
      const user = {
        name: "Ram贸n",
        password: "damedecome",
        username: "elsithecroc",
        photo: "",
        bio: "",
      };

      const userCard = ReactTestRenderer.create(<User user={user} />);
      expect(userCard.toJSON()).toMatchSnapshot();
    });
  });
});
