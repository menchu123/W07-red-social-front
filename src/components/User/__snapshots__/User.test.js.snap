// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a User component When it recives and object then it should render a card with the object info inside 1`] = `
<section
  className="user"
>
  <div
    className="user__top"
  >
    <img
      alt="Ramón"
      className="user__photo"
      height="150"
      src="https://i.scdn.co/image/ab67616d0000b2736f79fa715039164d02ddf1a8"
      width="150"
    />
    <div
      className="user__data"
    >
      <div
        className="user__names"
      >
        <span
          className="user__name"
        >
          Ramón
        </span>
        <span
          className="user__username"
        >
          @
          elsithecroc
        </span>
      </div>
      <div
        className="friend-buttons"
      >
        <button
          className="friend-button friend-button--friend"
        >
          :)
        </button>
        <button
          className="friend-button friend-button--enemy"
        >
          :(
        </button>
      </div>
    </div>
  </div>
  <div
    className="user__bio"
  >
    Hello, I'm new at Frenemies!
  </div>
</section>
`;
